### PR TITLE
EN-8500: ESDT fixes

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Added the number of decimals when issuing a new ESDT token. It should be the 4th parameter, right after the initial supply. Also exported it in the view function

### CODE REVIEW

File: vm/systemSmartContracts/esdt.go [hunk 1]
```diff
@@ -254,6 +257,8 @@ func (e *esdt) createNewTokenIdentifier(caller []byte, ticker []byte) ([]byte, e
 }
 
 func (e *esdt) issueToken(owner []byte, arguments [][]byte) error {
+	// format: issue@tokenName@ticker@initialSupply@numOfDecimals@optional-list-of-properties
```
Comment (sasurobert) [inline vm/systemSmartContracts/esdt.go:260]: move before the function

**Review by sasurobert — APPROVED** [review]

File: vm/systemSmartContracts/esdt.go (hunk 1 above)
Comment (bogdan-rosianu) [reply vm/systemSmartContracts/esdt.go]: done

**Review by SebastianMarian — APPROVED** [review]

File: process/smartContract/process.go [hunk 2]
```diff
@@ -894,7 +894,7 @@ func (sc *scProcessor) isCrossShardESDTTransfer(tx data.TransactionHandler) bool
 		return false
 	}
 
-	return function == core.BuiltInFunctionESDTTransfer
+	return function == core.BuiltInFunctionESDTTransfer || function == core.BuiltInFunctionESDTBurn
```
Comment (sasurobert) [inline process/smartContract/process.go:897]: delete this :D not needed

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: integrationTests/vm/esdt/esdtProcess_test.go [hunk 3]
```diff
@@ -257,6 +375,12 @@ func checkAddressHasESDTTokens(
 	value *big.Int,
 ) {
 	esdtData := getESDTTokenData(t, address, nodes, tokenName)
+	/* useful prints
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply integrationTests/vm/esdt/esdtProcess_test.go]: done

File: process/smartContract/process.go (hunk 2 above)
Comment (bogdan-rosianu) [reply process/smartContract/process.go]: done

**Review by SebastianMarian — APPROVED** [review]

**Review by SebastianMarian — DISMISSED** [review]

File: process/smartContract/builtInFunctions/esdtFreezeWipe.go [hunk 4]
```diff
@@ -85,6 +85,20 @@ func (e *esdtFreezeWipe) ProcessBuiltinFunction(
...
+	if err != nil {
+		return err
+	}
+
+	esdtUserMetadata := ESDTUserMetadataFromBytes(tokenData.Properties)
+	if !esdtUserMetadata.Frozen {
+		return process.ErrCannotWipeAccountNotFrozen
+	}
+
+	return acntDst.DataTrieTracker().SaveKeyValue(tokenKey, nil)
```
Comment (sasurobert) [inline process/smartContract/builtInFunctions/esdtFreezeWipe.go:99]: we should resolve to not send back this data to the metachain -> does we can have a silent error here. with returnMessage -> cannot wipe as it was not freezed.

Comment (sasurobert) [reply process/smartContract/builtInFunctions/esdtFreezeWipe.go:99]: plus add code in ProcessIferror to not Add the created SCR if shardID := sc.shardCoordinator.ComputeId(scr.GetRcvAddr())
if shardID == core.MetachainShardId && scr.GetGasLimit() == 0 && scr.GetValue().Cmp(zero) == 0 {
   continue
}

**Review by sasurobert — CHANGES_REQUESTED** [review]

**Review by SebastianMarian — DISMISSED** [review]

File: process/smartContract/builtInFunctions/esdtFreezeWipe.go (hunk 4 above)
Comment (sasurobert) [reply process/smartContract/builtInFunctions/esdtFreezeWipe.go:99]: my suggestion does not work - it can remain as it is. it works fine.

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**